src/engine/input/key_mapping/InputEventItem.cs

```diff
@@ -132,6 +147,83 @@ public override void _ExitTree()
         KeyPromptHelper.ControllerTypeChanged -= OnControllerChanged;
     }
 
+    public override void _Process(double delta)
+    {
+        // Delayed key binding (to allow shift to be used in modifier combinations, for example)
+        if (modifierKeyStatus == ModifierKeyMode.None)
+            return;
+
+        timeSinceModifier += delta;
```

Comment:
I noticed that with standalone Shift keybind there is delay between when button is pressed and when it is accepted, it is minor thing though and I guess sooner or later player would figure it out

Reply:
It's kind of necessary because otherwise you cannot bind stuff to like "Shift+W" as it would immediately just bind to shift and the W input would be just ignored. So there has to be some delay, though if lower value still works that could be fine (like maybe 300 ms or so).